edit/edit.go

```diff
@@ -711,3 +748,21 @@ func parseMarkRune(cmd []rune) (rune, error) {
 	}
 	return ' ', errors.New("bad mark: " + string(cmd[i]))
 }
+
+// parseNumber parses and returns a positive integer. The first returned
+// value is the parsed number, the second is the number of runes parsed.
+func parseNumber(cmd []rune) (int, []rune, error) {
+	i := 0
+	n := 1 // by default use the first instance
+	var err error
+	for len(cmd) > i+1 && unicode.IsDigit(cmd[i+1]) {
```

Comment:
I think this is assuming that cmd begins with 's' (right?). It shouldn't. It should expect cmd to begin with a number, and the caller should strip the 's' by passing cmd[1:].